src/CompoundRegion.cc

```diff
@@ -61,28 +61,38 @@ std::uint64_t consumeDecodeU64(std::uint8_t const *&buffer, std::uint8_t const *
 template <typename F>
 auto getUnionBounds(UnionRegion const &compound, F func) {
     auto bounds = func(compound.getOperand(0));
-    bounds.expandTo(func(compound.getOperand(1)));
+    for (unsigned i = 1; i < compound.nOperands(); ++ i) {
+        bounds.expandTo(func(compound.getOperand(i)));
+    }
     return bounds;
 }
 
 template <typename F>
 auto getIntersectionBounds(IntersectionRegion const &compound, F func) {
     auto bounds = func(compound.getOperand(0));
-    bounds.clipTo(func(compound.getOperand(1)));
+    for (unsigned i = 1; i < compound.nOperands(); ++ i) {
+        bounds.clipTo(func(compound.getOperand(i)));
+    }
     return bounds;
 }
 
 }  // namespace
 
-CompoundRegion::CompoundRegion(Region const &first, Region const &second)
-        : _operands{first.clone(), second.clone()} {}
-
-CompoundRegion::CompoundRegion(
-    std::array<std::unique_ptr<Region>, 2> operands) noexcept
-        : _operands(std::move(operands)) {}
+CompoundRegion::CompoundRegion(std::vector<std::unique_ptr<Region>> operands) noexcept
+        : _operands(std::move(operands))
+{
+    if (_operands.empty()) {
+        throw std::invalid_argument("CompoundRegion requires non-empty region list.");
+    }
```

Comment:
An alternative to banning empty lists that might be more convenient would be to declare:

- an empty `UnionRegion` is an empty region, i.e. contains no points;
- an empty `IntersectionRegion` is a full region, i.e. contains the full sphere.

I think that's the natural limiting behavior for all of the `isDisjoint` and `relate` implementations.

Reply:
I actually started with that approach but it requires extra logic in few places which was not quite obvious to me, and I was not sure that it was useful at all. I'll try to restore that and add few unit tests.